### PR TITLE
Migrations

### LINKED ISSUE #24: Add columns to Subscriptions table
As a developer, I update the Subscriptions table according to the ERD:

- [ ] ADD next available date - Datetime

Also, add the following relationships:

- [ ] Users have many subscriptions
- [ ] Subscription belongs to a user
- [ ] Crates have many subscriptions
- [ ] Subscription belongs to a crate

Update the types.js and model.js files to reflect the new columns

* allowNull: false --> field cannot be null
* little note/reminder: check to ensure your migration has an up and a down

### LINKED ISSUE #25: Create Product History table
Create a product history table and add the following columns:

- [ ] id
- [ ] userID
- [ ] productID
- [ ] kept?

Update the types.js and model.js files to reflect the new columns

* allowNull: false --> field cannot be null
* little note/reminder: check to ensure your migration has an up and a down

### LINKED ISSUE #26: Create Crate Products table
Create a crate products table and add the following columns:

- [ ] id
- [ ] crateID
- [ ] productID

Update the types.js and model.js files to reflect the new columns

### PR DESCRIPTION
**Pull request by nicorithner**
Tasks:

  - Created productHistory.js
  - Created table crateProducts.js
  - Added nextDeliveryDate column to subscriptions table.

### CODE REVIEW

File: code/api/src/migrations/6-crateProducts.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
```
Comment (nicolegooden) [inline code/api/src/migrations/6-crateProducts.js:36]: This syntax looks so similar to `Knex.js`.  You should check it out sometime!

File: code/api/src/migrations/8-updateSubscriptions.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (nicolegooden) [inline code/api/src/migrations/8-updateSubscriptions.js:1]: What does `'use strict'` do?

Comment (ckccameron) [reply code/api/src/migrations/8-updateSubscriptions.js:1]: Hey Nicole - https://stackoverflow.com/questions/1335851/what-does-use-strict-do-in-javascript-and-what-is-the-reasoning-behind-it

This resource is what we used to answer that question. During our research into Node.js and how to set up our files, we kept seeing the `'use strict';` syntax at the top of files so we assumed that was something we had to include.

**Review by ckccameron — APPROVED** [review]
Looks good 🙌🏽 feel solid about how to create migrations with an up function as well as how to implement the down function.

**Pull request merged**